src/index/TextScoringEnum.cpp

```diff
@@ -7,29 +7,27 @@
 // ____________________________________________________________________________
 std::string getTextScoringMetricAsString(TextScoringMetric textScoringMetric) {
   switch (textScoringMetric) {
-    case TextScoringMetric::EXPLICIT:
+    using enum TextScoringMetric;
+    case EXPLICIT:
       return "explicit";
-      break;
-    case TextScoringMetric::TFIDF:
+    case TFIDF:
       return "tf-idf";
-      break;
-    case TextScoringMetric::BM25:
+    case BM25:
       return "bm25";
-      break;
     default:
-      return "count";
-      break;
+      return "explicit";
   }
 }
 
 // ____________________________________________________________________________
 TextScoringMetric getTextScoringMetricFromString(
-    std::string textScoringMetricString) {
+    const std::string& textScoringMetricString) {
+  using enum TextScoringMetric;
   if (textScoringMetricString == "tf-idf") {
-    return TextScoringMetric::TFIDF;
-  } else if (textScoringMetricString == "bm25") {
-    return TextScoringMetric::BM25;
-  } else {
-    return TextScoringMetric::EXPLICIT;
+    return TFIDF;
+  }
+  if (textScoringMetricString == "bm25") {
+    return BM25;
   }
+  return EXPLICIT;
```

Comment:
Shouldn't this have some kind of check that the string in the else case is "explicit", s.t. not "tfIdf" becomes "EXPLICIT", which would be unexpected.